src/FffGcodeWriter.cpp

```diff
@@ -1211,6 +1211,356 @@ void FffGcodeWriter::processSpiralizedWall(const SliceDataStorage& storage, Laye
+                    // move to the location of the vertex in the outermost enclosing inset that's closest to the z seam location
+                    const Point dest = hole_inner_walls.back()[PolygonUtils::findNearestVert(z_seam_location, hole_inner_walls.back())];
+                    gcode_layer.addTravel(dest);
+                    if (outer_poly_order_idx == 0)
+                    {
+                        // FIXME: adding the above travel move for the first hole in the part triggers an assertion unless we follow it with this bogus zero length extrude
+                        gcode_layer.addExtrusionMove(dest, &mesh_config.insetX_config, SpaceFillType::Polygons, flow, spiralize);
+                    }
+                }
+                std::reverse(hole_inner_walls.begin(), hole_inner_walls.end());
+                if (compensate_overlap_x)
+                {
+                    WallOverlapComputation wall_overlap_computation(hole_inner_walls, wall_line_width_x);
+                    gcode_layer.addPolygonsByOptimizer(hole_inner_walls, &mesh_config.insetX_config, &wall_overlap_computation);
+                }
+                else
+                {
+                    gcode_layer.addPolygonsByOptimizer(hole_inner_walls, &mesh_config.insetX_config);
+                }
+                if (compensate_overlap_0)
+                {
+                    WallOverlapComputation wall_overlap_computation(hole_outer_wall, wall_line_width_0);
+                    gcode_layer.addPolygonsByOptimizer(hole_outer_wall, &mesh_config.inset0_config, &wall_overlap_computation, z_seam_type, z_seam_pos, wall_0_wipe_dist, spiralize, flow, retract_before_outer_wall);
+                }
+                else
+                {
+                    WallOverlapComputation* wall_overlap_computation(nullptr);
+                    gcode_layer.addPolygonsByOptimizer(hole_outer_wall, &mesh_config.inset0_config, wall_overlap_computation, z_seam_type, z_seam_pos, wall_0_wipe_dist, spiralize, flow, retract_before_outer_wall);
+                }
+            }
+        }
+        else
+        {
+            // just the outer wall, no level 1 inset
+            if (compensate_overlap_0)
+            {
+                WallOverlapComputation wall_overlap_computation(hole_outer_wall, wall_line_width_0);
+                gcode_layer.addPolygonsByOptimizer(hole_outer_wall, &mesh_config.inset0_config, &wall_overlap_computation, z_seam_type, z_seam_pos, wall_0_wipe_dist, spiralize, flow, retract_before_outer_wall);
+            }
+            else
+            {
+                WallOverlapComputation* wall_overlap_computation(nullptr);
+                gcode_layer.addPolygonsByOptimizer(hole_outer_wall, &mesh_config.inset0_config, wall_overlap_computation, z_seam_type, z_seam_pos, wall_0_wipe_dist, spiralize, flow, retract_before_outer_wall);
+            }
+        }
+    }
+    // now process the part's outer wall and the level 1 insets that it surrounds
```

Comment:
This function is becoming very long. Please try to split it up into smaller parts to make the code more readable.